### PR TITLE
Minigame System - Base Functionality Complete

### PR DESCRIPTION
**Pull request by jurcaua**
Lots of things going on here, definitely need some eyes on this.

* 2 example minigames
  * ExampleMinigame showcases just a running timer dictating how much stamina is gained:
    * 3 possible modes currently supported:
      * Stamina decreases by some value (such as a minigame that requires speed)
      * Stamina increases by some value (such as a survive as long as you can minigame)
      * Stamina decreases within some time limit (same as the first point)
    * image
  * ExampleMinigameTarget showcases a segmented system where as the player completes a target, segments of the stamina bar will fill:
  * The number of segments is customizable, and the Sprite being used can be changed as well per minigame.
  * image
  * image
    * The `MinigameBarSetup` class will automatically take care of naming, required components, sprites, etc. (just duplicating any of the images or deleting and then saving will trigger it to clean everything up).

Player test animations added, and hooked up with a Cinemachine State Driven Camera
* this will transition between cameras based on an Animator component, and therefore when we go into the "Rest" animation state, the camera will then smoothly zoom to the player's head and start a random minigame.

Stamina Bar bugs fixed:
* Changes all to be used `Time.fixedDeltaTime` to fix inconsistent changes
  * #16 
* Now changes based on a fixed delta per coroutine rather than a "goal percentage" to reach, which led to issues when multiple corountines were running at the same time.

### CODE REVIEW

File: Assets/Scripts/Minigames/MinigameBarSetup.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        {
+            childImages.Clear();
+            foreach (Transform child in staminaFillTransform)
+            {
+                // because we cannot destroy objects in OnValidate, we wait until the end of the frame for this:
+                // https://answers.unity.com/questions/1318576/destroy-child-objects-onvalidate.html
+                StartCoroutine(DestroyAtEndOfFrame(child.gameObject));
+            }
+        }
+        else
```
Comment (jurcaua) [inline Assets/Scripts/Minigames/MinigameBarSetup.cs:47]: Any ideas on cleaning this code would be helpful. It's basically attempting to cover all cases when people can delete or add children to this stamina bar transform.

**Comment by jurcaua** [comment]
Newest push here is just making sure `AIScene` is updated as well (since I was just working in my own `Minigame_Testing` scene).

Also made prefabs of all the main game systems. Led to some issues with the newest code here so I had to add a bit more code to fix that (related to always being executed with default values first because of a weird bug with having the `[ExecuteAlways]` above the `MinigameBarSetup` class).

**Comment by jurcaua** [comment]
@justineechen helped me simplify the script I mentioned previously, so now it's a lot more simple.
* Before we used 3 separate sprites for the segmented stamina bar, but now we just use 1, making the logic for keeping it all consistent a lot more simple & maintainable.

**Review by chadleytan — COMMENTED** [review]
Looks good to me!

**Pull request merged**